test/hooks_test.lua

```diff
@@ -311,3 +311,25 @@ g.test_each_error = function()
     t.assert_equals(result, 1)
     t.assert_equals(hooks, {'before_each1', 'after_each1'})
 end
+
+g.test_each_repeat = function()
+    local hooks = {}
+
+    local result = helper.run_suite(function(lu2)
+        local t2 = lu2.group('test')
+        t2.before_each(function() table.insert(hooks, 'before_each') end)
+        t2.before_test('test_a', function() table.insert(hooks, 'before_test') end)
+        t2.after_each(function() table.insert(hooks, 'after_each') end)
+        t2.after_test('test_b', function() table.insert(hooks, 'after_test') end)
+        t2.test_a = function() table.insert(hooks, 'test_a') end
+        t2.test_b = function() table.insert(hooks, 'test_b') end
+    end, {'--repeat', '2'})
+
+    t.assert_equals(result, 0)
+    t.assert_equals(hooks, {
+        'before_each', 'before_test', 'test_a', 'after_each',
+        'before_each', 'before_test', 'test_a', 'after_each',
+        'before_each', 'test_b', 'after_test', 'after_each',
+        'before_each', 'test_b', 'after_test', 'after_each',
+    })
```

Comment:
Would you like to extend it with before_all and after_all?

Also, are there tests that cover:
- an error in `before_each` in conjunction with `--repeat` option
- a case when n-th test fails

Reply:
> Would you like to extend it with before_all and after_all?

I don't think that repeating `_all` hooks is a good idea. They are supposed to be before and after **all** tests (evidently), but with repeat option it won't be this way.

Reply:
I didn't mean calling `before_all` twice but extending the test with it. Never mind, you can merge it as is.

Reply:
My bad. Added it to the test